### ISSUE TITLE
Traefik v2.4.11 respond with 404 for more than 5 minutes after load test with k6

### ISSUE THREAD

**Issue by Nadavtalon** [traefik/traefik#9041]
### Welcome!

- [X] Yes, I've searched similar issues on [GitHub](https://github.com/traefik/traefik/issues) and didn't find any.
- [X] Yes, I've searched similar issues on the [Traefik community forum](https://community.traefik.io) and didn't find any.

### Scenario
I created several deployments of my application (with service of `type: ClusterIP` and ingress `networking.k8s.io/v1`) in GKE private cluster `v1.21.10-gke.2000`. 
My app runs in a pod behind traefik v2.4.11 and one of my deployments was set to fail and cause container restarts (the application pod will be in `CrashLoopBackOff` as soon as the db connections limit will exceed).
The deployment of the application uses horizontal pod autoscaler `autoscaling/v1` and scale up and down by cpu % target.
I run `k6` to send big amount of requests to the external load balancer DNS which is my traefik service ip. `k6` run from several locations (one of them is from a pod inside the cluster, others from external locations).

### What happened
After running load tests for several minutes and by increasing the amount of requests, traefik responded with few 502 and 500 errors (as expected) but then started to send 404 and non of the application deployment (even deployments which were not part of the load test) were accessible or from k6 or from a browser - all responded with 404.
Traefik was not able to recover for about 5 minutes or more and return 404 while trying to access all of the deployments behind traefik.

### Application description
The application runs ReactJS in frontend and Golang in the backend from kubernetes GKE private cluster pod (one pod running frontend and backend together) and connected to postgres database in internal VPC of the db service.

### Notes
- I run the same scenario with traefik v2.6.3 and got the same 404 errors
- I run the same scenario with traefik v1.7.30 and got only 502/503 errors (as expected)
- I reproduce this event several times in 2 GKE clusters (one was completely new cluster created with terraform)
- While reproducing in a dedicated new cluster it happened only after I run k6 from inside the cluster, but the cluster was bigger and similar to our production environment
- In another time I reproduce the issue, some deployments behind traefik returned 404 without having any internal server errors or service unavailable
- CPU \Memory usage of traefik pods was stable and didn't exceed limits

### What did you see instead?

I expected to see that traefik v2 respond with 502/503 errors as my containers restarted and making the service unavailable, but instead I saw few 500 (expected because of db connection limits) few 502 (expected because of container restarts) but instead I got 404 for all the deployments that connected to the same database (in some cases it was for all the deployments behind traefik).

There is a related discussion: https://github.com/traefik/traefik/issues/8141

### What version of Traefik are you using?

Currently we are using `traefik` for Kubernetes ingress with `image: traefik:v2.4.11`

I run the same scenario with `image: traefik:v1.7.30-alpine` and with `image: traefik:v2.6.3` (This bug \ config error happened in both `v2.4.11` and `v2.6.3` but not with `v1.7.30-alpine`.

### What is your environment & configuration?

Traefik config toml static and dynamic files from a configMap:
```toml
  static.toml: |
    [log]
      level = "INFO"
    [accessLog]

    [providers]
      [providers.file]
        filename = "/config/dynamic.toml"
        watch = true
      [providers.kubernetesIngress]
        namespaces = []

    [metrics]
      [metrics.datadog]
        address = "datadog-agent.datadog.svc.cluster.local:8125"
        pushInterval = "10s"
    [entryPoints]
      [entryPoints.web]
        address = ":80"
        [entryPoints.web.http]
          [entryPoints.web.http.redirections]
            [entryPoints.web.http.redirections.entryPoint]
            to = "websecure"
            scheme = "https"
        [entryPoints.web.transport]
          [entryPoints.web.transport.lifeCycle]
            requestAcceptGraceTimeout = 42
            graceTimeOut = 42   
      [entryPoints.websecure]
        address = ":443"
        [entryPoints.websecure.http]
          [entryPoints.websecure.http.tls]
        [entryPoints.websecure.transport]
          [entryPoints.websecure.transport.lifeCycle]
            requestAcceptGraceTimeout = 42
            graceTimeOut = 42 
      [entryPoints.ping]
        address = ":8082"

      [ping]
        entryPoint = "ping"
        terminatingStatusCode = 204

  dynamic.toml: |

    [http]
      [http.routers]
        [http.routers.web]
          entryPoints = [ "web" ]
          service = "lb"
          rule = "Host(`example.com`)"
        [http.routers.websecure]
          entryPoints = [ "websecure" ]
          service = "lb"
          rule = "Host(`example.com`)"
          [http.routers.websecure.tls]
            options = "myTLSOptions"

    [services]
      [http.services]
        [http.services.lb]
          [http.services.lb.loadBalancer.sticky.cookie]
            name = "my_sticky_cookie_name"
            secure = true
            httpOnly = true
            sameSite = "none"

    [tls]
      [[tls.certificates]]
        certFile = "/tls.crt"
        keyFile = "/tls.key"   

      [tls.stores]
        [tls.stores.default]
          [tls.stores.default.defaultCertificate]
            certFile = "/tls.crt"
            keyFile = "/tls.key" 

      [tls.options]
        [tls.options.myTLSOptions]
          minVersion = "VersionTLS12"
          cipherSuites = [
            "TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256",
            "TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384",
            "TLS_ECDHE_RSA_WITH_CHACHA20_POLY1305",
            "TLS_ECDHE_RSA_WITH_AES_128_CBC_SHA",
            "TLS_ECDHE_RSA_WITH_AES_256_CBC_SHA",
            "TLS_RSA_WITH_AES_128_GCM_SHA256",
            "TLS_RSA_WITH_AES_256_GCM_SHA384",
            "TLS_RSA_WITH_AES_128_CBC_SHA",
            "TLS_RSA_WITH_AES_256_CBC_SHA",
            "TLS_ECDHE_ECDSA_WITH_CHACHA20_POLY1305_SHA256",
            "TLS_ECDHE_RSA_WITH_CHACHA20_POLY1305_SHA256",
            "TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256",
          ]
```

Traefik deployment specifications in Kubernetes:
```yaml
apiVersion: apps/v1
kind: Deployment
metadata:
  labels:
    app: traefik
  name: traefik
  namespace: traefik-ingress
spec:
  replicas: 2
  selector:
    matchLabels:
      app: traefik
  strategy:
    rollingUpdate:
      maxSurge: 1
      maxUnavailable: 1
    type: RollingUpdate
  template:
    metadata:
      labels:
        app: traefik
    spec:
      containers:
      - image: traefik:v2.4.11
        name: traefik
        imagePullPolicy: IfNotPresent
        args:
          - --configfile=/config/static.toml
        ports:
        - containerPort: 80
          name: http
          protocol: TCP
        - containerPort: 443
          name: https
          protocol: TCP
        - containerPort: 8080
          name: dashboard
          protocol: TCP
        resources:
          limits:
            cpu: 1000m
            memory: 1024Mi
          requests:
            cpu: 200m
            memory: 256Mi
        livenessProbe:
          httpGet:
            path: /ping
            port: 8082
            scheme: HTTP
          initialDelaySeconds: 5
          timeoutSeconds: 2 
        terminationMessagePath: /dev/termination-log
        terminationMessagePolicy: File
        volumeMounts:
        - mountPath: /config
          name: config
        - mountPath: /certs
          name: certs
      dnsPolicy: ClusterFirst
      restartPolicy: Always
      schedulerName: default-scheduler
      securityContext: {}
      terminationGracePeriodSeconds: 90
      serviceAccountName: tld-traefik-ingress-controller
      volumes:
      - configMap:
          defaultMode: 420
          name: traefik-config
        name: config
      - name: certs
        secret:
          defaultMode: 420
          secretName: certs
---
apiVersion: v1
kind: Service
metadata:
  labels:
    app: traefik
  name: tld-traefik-ingress
  namespace: tld-traefik-ingress
spec:
  externalTrafficPolicy: Local
  ports:
  - name: http
    port: 80
    protocol: TCP
    targetPort: 80
  - name: https
    port: 443
    protocol: TCP
    targetPort: 443
  selector:
    app: traefik
  sessionAffinity: None
  type: LoadBalancer
  loadBalancerIP: 10.10.10.10
```

Add more configuration information here.

### If applicable, please paste the log output in DEBUG level

I could not collect relevant debug messages as k6 run too many requests to traefik \ my app. 

From traefik (these are just kind of error logs, but the amount is much higher):
```
"10.16.0.10 - - [16/May/2022:16:37:38 +0000] "PUT /v1/customer_sessions/c762c8b0-7ca2-440b-8ada-ede2d08f01de HTTP/2.0" 404 19 "-" "-" 189423 "-" "-" 0ms"
"87.183.173.151 - - [16/May/2022:16:26:43 +0000] "GET /login?next=%2F HTTP/2.0" 404 19 "-" "-" 276256 "-" "-" 0ms"
"35.190.223.150 - - [16/May/2022:10:35:02 +0000] "POST /v1/accounts HTTP/2.0" 404 19 "-" "-" 4249 "-" "-" 0ms"
```
From `k6` (again, the amount of logs is much higher):
```
time="2022-05-16T16:33:49Z" level=info msg="ERROR: 500 {\"message\":\"Internal server error\",\"errors\":null,\"StatusCode\":500}" source=console
time="2022-05-16T16:33:49Z" level=info msg="ERROR: 404 404 page not found\n" source=console
time="2022-05-16T16:33:49Z" level=info msg="ERROR: 404 404 page not found\n" source=console
...
time="2022-05-16T16:33:52Z" level=info msg="ERROR: 404 404 page not found\n" source=console
time="2022-05-16T16:33:52Z" level=info msg="ERROR: 500 {\"message\":\"Internal server error\",\"errors\":null,\"StatusCode\":500}" source=console
```

In traefik `v1.7.30`, while trying to reproduce the same scenario I got this (Expected):
```
time="2022-05-17T10:54:24Z" level=info msg="ERROR: valid_coupon 503 Service Unavailable" source=console
time="2022-05-17T10:54:24Z" level=info msg="ERROR: no_coupon 503 Service Unavailable" source=console
time="2022-05-17T10:54:25Z" level=error msg="TypeError: Cannot read property 'length' of undefined\n\tat coupon was accepted (file:///app/testcases/CouponCampaign.js:88:27(32))\n\tat go.k6.io/k6/js/modules/k6.(*K6).Check-fm (native)\n\tat file:///app/testcases/CouponCampaign.js:82:15(181)\n\tat native\n" executor=constant-vus scenario=default source=stacktrace
time="2022-05-17T10:54:25Z" level=error msg="TypeError: Cannot read property 'length' of undefined\n\tat coupon was accepted (file:///app/testcases/CouponCampaign.js:88:27(32))\n\tat go.k6.io/k6/js/modules/k6.(*K6).Check-fm (native)\n\tat file:///app/testcases/CouponCampaign.js:82:15(181)\n\tat native\n" executor=constant-vus scenario=default source=stacktrace
time="2022-05-17T10:54:25Z" level=info msg="ERROR: 503 Service Unavailable" source=console
```

**Comment by Nadavtalon** [comment]
Hi @rtribotte ,

I tried to set `allowEmptyServices` in traefik `v2.4.11` but I keep on getting this error:
```
$ k logs --previous traefik-6b6bfd4cd7-4ws5c
2022/05/31 13:44:10 command traefik error: field not found, node: allowEmptyServices
```
Other params are working fine (I try to add `throttleDuration` and `allowExternalNameServices` but `allowEmptyServices` is not working). This is my `static.toml` file:
```
    [log]
      level = "INFO"
    [accessLog]

    [providers]
      [providers.file]
        filename = "/config/dynamic.toml"
        watch = true
      [providers.kubernetesIngress]
        throttleDuration = "10s"
        namespaces = []

    [metrics]
      [metrics.datadog]
        address = "datadog-agent.datadog.svc.cluster.local:8125"
        pushInterval = "10s"
    [entryPoints]
      [entryPoints.web]
        address = ":80"
        [entryPoints.web.http]
          [entryPoints.web.http.redirections]
            [entryPoints.web.http.redirections.entryPoint]
            to = "websecure"
            scheme = "https"
        [entryPoints.web.transport]
          [entryPoints.web.transport.lifeCycle]
            requestAcceptGraceTimeout = 42
            graceTimeOut = 42   
      [entryPoints.websecure]
        address = ":443"
        [entryPoints.websecure.http]
          [entryPoints.websecure.http.tls]
        [entryPoints.websecure.transport]
          [entryPoints.websecure.transport.lifeCycle]
            requestAcceptGraceTimeout = 42
            graceTimeOut = 42 
      [entryPoints.ping]
        address = ":8082"

      [ping]
        entryPoint = "ping"
        terminatingStatusCode = 204
```

**Comment by Nadavtalon** [comment]
Hi @rtribotte !

Thank you for your message! I set my pods to `v2.5` and I get from my console 503 as expected:
```
INFO[0076] ERROR: 503 Service Unavailable                source=console
INFO[0076] ERROR: 503 Service Unavailable                source=console
INFO[0076] ERROR: 503 Service Unavailable                source=console
WARN[0076] Request Failed                                error="Put \"https://someURL/v2/customer_sessions/4489b92d-b53f-4ed0-8910-039a1b216ab4\": request timeout"
INFO[0076] ERROR: 0 null                                 source=console
INFO[0076] ERROR: 503 Service Unavailable                source=console
INFO[0076] ERROR: 503 Service Unavailable                source=console
```
The main issue right now is that only my `k6` client can see these 503 errors. From `traefik` pod the errors are not reported at all. Is there additional configuration to report these 503 errors?
